src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
```

Comment:
It's just for keep the simplicity.. We have two options:

1. Change to protected
2. Refactor `getBeanSession()` to `session()`

You choose.